src/main/java/org/sonarsource/sonarlint/ls/connected/ProjectBindingManager.java

```diff
@@ -437,19 +440,7 @@ public void forEachBoundFolder(BiConsumer<WorkspaceFolderWrapper, WorkspaceFolde
     });
   }
 
-  public void handleEvents(ServerEvent event) {
-    if (event instanceof TaintVulnerabilityRaisedEvent) {
-      var filePathFromEvent = ((TaintVulnerabilityRaisedEvent) event).getMainLocation().getFilePath();
-      var localFileUri = serverPathToFileUri(filePathFromEvent);
-      localFileUri.ifPresent(this::updateTaintIssueCacheFromStorageForFile);
-    } else if (event instanceof TaintVulnerabilityClosedEvent ||
-      event instanceof IssueChangedEvent) {
-      taintVulnerabilitiesCache.getAllFilesWithTaintIssues()
-        .forEach(this::updateTaintIssueCacheFromStorageForFile);
-    }
-  }
-
-  private void updateTaintIssueCacheFromStorageForFile(URI fileUri) {
+  public void updateTaintIssueCacheFromStorageForFile(URI fileUri) {
```

Comment:
event handler moved to a separate class. This method made public so that it is accessible for event handler